Doc/howto/regex.rst

```diff
@@ -549,27 +545,39 @@ of each one.
 
    Perform case-insensitive matching; character class and literal strings will
    match letters by ignoring case.  For example, ``[A-Z]`` will match lowercase
-   letters, too, and ``Spam`` will match ``Spam``, ``spam``, or ``spAM``. This
-   lowercasing doesn't take the current locale into account; it will if you also
-   set the :const:`LOCALE` flag.
+   letters, too. Full Unicode matching also works unless the :const:`re.ASCII`
+   flag is also used to disable non-ASCII matches.  ``[A-Z]`` will also match
+   letters 'İ' (U+0130, Latin capital letter I with dot above), 'ı' (U+0131,
+   Latin small letter dotless i), 'ſ' (U+017f, Latin small letter long s) and
+   'K' (U+212a, Kelvin sign) in Unicode mode.  ``Spam`` will match ``Spam``,
+   ``spam``, ``spAM``, or ``ſpam`` (the latter is matched only in Unicode mode).
+   This lowercasing doesn't take the current locale into account; it will if
+   you also set the :const:`LOCALE` flag.
 
 
 .. data:: L
           LOCALE
    :noindex:
 
-   Make ``\w``, ``\W``, ``\b``, and ``\B``, dependent on the current locale
-   instead of the Unicode database.
-
-   Locales are a feature of the C library intended to help in writing programs that
-   take account of language differences.  For example, if you're processing French
-   text, you'd want to be able to write ``\w+`` to match words, but ``\w`` only
-   matches the character class ``[A-Za-z]``; it won't match ``'é'`` or ``'ç'``.  If
-   your system is configured properly and a French locale is selected, certain C
-   functions will tell the program that ``'é'`` should also be considered a letter.
+   Make ``\w``, ``\W``, ``\b``, ``\B`` and case-insensitive matching dependent
+   on the current locale instead of the Unicode database.
+
+   Locales are a feature of the C library intended to help in writing programs
+   that take account of language differences.  For example, if you're
+   processing encoded French text, you'd want to be able to write ``\w+`` to
+   match words, but ``\w`` only matches the character class ``[A-Za-z]`` in
+   bytes patterns; it won't match bytes corresponding to ``'é'`` or ``'ç'``.
```

Comment:
It is not clear why b'\xe9' and b'\xe7' should be matched. But 'é' and 'ç' are French letters, and I have added "bytes corresponding to" for making this phrase Python 3 compatible.

Reply:
The reason why I suggested that, is because ``'é'`` and ``'ç'`` are Unicode str in Python 3, whereas without quotes they are just letters.  The addition of ``b'\xe9'`` and ``b'\xe7'`` might help clarify what is being matched, but it's not essential.